src/infra/communication/connect.ts

```diff
@@ -0,0 +1,98 @@
+import assert from 'assert'
+import { Subject } from './subject'
+import { Observer } from './observer'
+
+type Resolver<T> = Record<string, (...args: any[]) => T | T[] | undefined>
+type Mutator = Record<string, (...args: any[]) => void>
+
+export interface WebSocketServer {
+  on(event: 'connection', cb: (socket: WebSocket) => void): void
+}
+
+export interface WebSocket {
+  send(payload: string): void
+  on(event: 'message', cb: (message: string) => void): void
+  once(event: 'close', cb: () => void): void
+}
+
+export interface ConnectConfig<T> {
+  resolver: Resolver<T>
+  mutator: Mutator
+  subject: Subject<T>
+  server: WebSocketServer
+}
+
+export function connectWsServer<T>({
+  resolver,
+  mutator,
+  subject,
+  server
+}: ConnectConfig<T>): void {
+  server.on('connection', ws => {
+    connectSubject(ws, subject)
+    listenMessage(ws, resolver, mutator)
+  })
+}
+
+function listenMessage(
+  ws: WebSocket,
+  resolver: Resolver<{}>,
+  mutator: Mutator
+) {
+  ws.on('message', message => {
+    const data = JSON.parse(message.toString())
+    const [type, method] = data.type.split(':')
+    const target = type === 'resolver' ? resolver : mutator
+
+    assert(type === 'resolver' || type === 'mutator')
+    assert(Array.isArray(data.args))
+    assert(typeof data.requestId === 'string')
```

Comment:
This may not need to be a string value. A number value is also fine.
It would be more convenient if this assertion only checks the property existence.